gateway/conf.d/apicast.conf

```diff
@@ -75,6 +75,29 @@ location @upstream {
   proxy_set_header X-3scale-debug "";
   proxy_set_header Connection "";
 
+  # This is a bit tricky. It uses liquid to set a SSL client certificate. In
+  # NGINX, all this is not executed as it is commented with '#'. However, in
+  # Liquid, all this will be evaluated. As a result, the following directives
+  # are set optionally: proxy_ssl_certificate, proxy_ssl_certificate_key,
+  # and proxy_ssl_session_reuse. The last one is 'on' by default, and I found
+  # that the certificate defined in this location was used in other ones.
+  # Setting it to 'off' fixes the issue.
+
+  # {% if proxy_ssl_certificate and proxy_ssl_certificate_key %}
+  #   {% capture proxy_ssl_session_reuse %}
+  #   proxy_ssl_session_reuse off;
+  #   {% endcapture %}
+  #   {% capture ssl_certificate %}
+  #   proxy_ssl_certificate {{ proxy_ssl_certificate }};
+  #   {% endcapture %}
+  #   {% capture ssl_certificate_key %}
+  #   proxy_ssl_certificate_key {{ proxy_ssl_certificate_key }};
```

Comment:
Lets not forget [proxy_ssl_password_file](http://nginx.org/en/docs/http/ngx_http_proxy_module.html#proxy_ssl_password_file).

Reply:
Yes. Otherwise, we'll need to type the password when Apicast boots.